### PR TITLE
Fix distributed log

### LINKED ISSUE #838: Logging with multiple processes duplicates info
e.g. https://buildkite.com/clima/climaatmos-scaling/builds/252#018335a9-1ee5-4adf-9eb9-d74f789b3aff

Probably needs some rearranging of the driver file

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR fixes the log for distributed runs by moving the distributed code closer up to the top of the driver.

Closes #838.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
Sorry, this doesn't seem to be fixing the issue yet!

File: examples/hybrid/driver.jl [hunk 1]
```diff
@@ -3,18 +3,28 @@ if !(@isdefined parsed_args)
...
+ClimaComms.init(comms_ctx)
+
 import Logging
-atexit() do
-    Logging.global_logger(Logging.global_logger(get_logger(comms_ctx)))
+if ClimaComms.iamroot(comms_ctx)
+    Logging.ConsoleLogger(stderr, Logging.Info)
+else
+    Logging.NullLogger()
+end
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:23]: What is this doing? It doesn't seem like the logger is being assigned to any variables or used anywhere

Comment (sriharshakandala) [reply examples/hybrid/driver.jl:23]: Can we pass in communicator `context` as an input argument to the hybrid driver?

Comment (simonbyrne) [reply examples/hybrid/driver.jl:23]: oh right

**Comment by charleskawczynski** [comment]
Looks like it's working, thanks for the fixes @simonbyrne!

**Comment by charleskawczynski** [comment]
I'll squash and merge

**Pull request merged**